ivc/src/poseidon/interpreter.rs

```diff
@@ -28,6 +28,48 @@ pub trait Params<F: PrimeField, const STATE_SIZE: usize, const NB_FULL_ROUNDS: u
     fn mds(&self) -> [[F; STATE_SIZE]; STATE_SIZE];
 }
 
+/// Populates and checks one poseidon invocation.
+pub fn poseidon_circuit<
+    F: PrimeField,
+    const STATE_SIZE: usize,
+    const NB_FULL_ROUND: usize,
+    PARAMETERS,
+    Env,
+>(
+    env: &mut Env,
+    param: PARAMETERS,
+    init_state: (F, F, F),
```

Comment:
That should be a list of the state size to stay consistent with the whole design.